### PR TITLE
feat: implement habit creation with icons and settings

### PR DESCRIPTION
**Pull request by owengretzinger**
### TL;DR

Added habit creation functionality with customizable icons, colors, and settings.

I also made it so that instead of explicitly setting a goal for how many times you want to complete the habit a day, there's just a setting for allowing multiple completions per day vs only allowing one. (See previous PR #145.) I implemented this functionality in this PR.

https://github.com/user-attachments/assets/e67199aa-9c44-4b33-a468-d74432e72225

### What changed?

- Created a new habit creation form with fields for title, description, icon selection, and color
- Added a modal for selecting from 100+ habit icons
- Implemented toggle for allowing multiple completions per day
- Updated habit schemas to support new icon system and multiple completions setting
- Replaced text emoji icons with Lucide icons throughout the app
- Reorganized habit color definitions and added color picker UI
- Added the checkbox.tsx component from the Obytes template which includes a switch that I used for toggling multiple completions per day

### How to test?

1. Navigate to the habits screen
2. Tap the "+" button to create a new habit
3. Fill out the habit creation form:
   - Select an icon from the icon picker
   - Enter a title and optional description
   - Choose a color from the color grid
   - Toggle "Allow multiple completions per day" setting
4. Create the habit and verify it appears in the habits list
5. Test the multiple completions setting by attempting to complete the habit multiple times in one day

### CODE REVIEW

**Comment by owengretzinger** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/habits-together/app/146?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#146** <a href="https://app.graphite.dev/github/pr/habits-together/app/146?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈 <a href="https://app.graphite.dev/github/pr/habits-together/app/146?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* **#145** <a href="https://app.graphite.dev/github/pr/habits-together/app/145?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#144** <a href="https://app.graphite.dev/github/pr/habits-together/app/144?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `main`


<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.

**Review by AnkushSarkar10 — DISMISSED** [review]

**Review by AnkushSarkar10 — APPROVED** [review]
